pymodbus/client/tcp.py

```diff
@@ -102,7 +102,7 @@ def __init__(
         host: str,
         port: int = 502,
         framer: Type[ModbusFramer] = ModbusSocketFramer,
-        source_address: Tuple[str, int] = None,
+        source_address: Optional[Tuple[str, int]] = None,
```

Comment:
Same here, please do not mix "Union", "Optional" and "|", that makes the code more complicated than needed.